compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -851,6 +851,31 @@ static void PrintMethodHandlerClass(const ServiceDescriptor* service,
   p->Print("}\n\n");
 }
 
+static void PrintGetServiceDescriptorMethod(const ServiceDescriptor* service,
+                                   map<string, string>* vars,
+                                   Printer* p,
+                                   bool generate_nano) {
+  (*vars)["service_name"] = service->name();
+  p->Print(
+      *vars,
+      "public static $ServiceDescriptor$ getServiceDescriptor() {\n");
```

Comment:
Committing to this doesn't seem that onerous. I find it hard to see making this instance based rather than static but it's possible I suppose.